### ISSUE TITLE
object_usage_linter ignores library functions

### ISSUE THREAD

**Issue by renkun-ken** [r-lib/lintr#482]
From https://github.com/REditorSupport/vscode-r-lsp/issues/36.

Suppose the following code is stored in `test.R`:

```r
library(jsonlite)

read <- function(file, ...) {
  read_json(file, ...)
}
```

Then `object_usage_linter` seems to ignore `library(jsonlite)` and mark `read_json` as unknown which is annoying.

```
> lintr::lint("test.R")
/Users/ken/Workspaces/test/vscode-r/test/test.R:4:3: warning: no visible global function definition for 'read_json'
  read_json(file, ...)
  ^~~~~~~~~
```

**Comment by tzakharko** [comment]
I was about to open an identical issue, so instead I am commenting here. In addition, maybe one should reconsider enabling `object_usage_linter` as default? If I understand correctly, it executes code during checks and is therefore a security hazard.

**Comment by assignUser** [comment]
This also happens if it is just in another .R file, it does *not* happen if the call is directly preceeded by an `if` *without* curly braces:
![image](https://user-images.githubusercontent.com/16141871/90273060-a2edb180-de5e-11ea-9626-7786e6eef391.png)

**Comment by naupaka** [comment]
Adding my +1 to this. It looks like the environment in which the code from outside the function definition is evaluated is not preserved when evaluating code that makes up the function definition. If the file could be parsed and the package variables added to the environment used for testing here:
https://github.com/jimhester/lintr/blob/b93765553d3c1194e37be6efa047085de6d06d5a/R/object_usage_linter.R#L22

that might be one way to resolve this? I tried fiddling with it, but I'm not comfortable enough with the structure of the package to know precisely where the leverage point would be.

**Comment by AshesITR** [comment]
Note that a somewhat hacky work-around is to source the file before linting / manually attach all needed packages.
This of course isn't ideal and doesn't work reliably in non-interactive sessions.

It works because the default environment in non-packages is `globalenv()`, where the manual changes can be seen by lintr.

**Comment by AshesITR** [comment]
This has been fixed.
